target_test.go

```diff
@@ -392,3 +393,86 @@ func TestQueryHandlerODoHWithCorruptCiphertext(t *testing.T) {
 		t.Fatal(fmt.Errorf("Result did not yield %d, got %d instead", http.StatusBadRequest, status))
 	}
 }
+
+func TestQueryHandlerODoHWithMalformedQuery(t *testing.T) {
+	r := createLocalResolver(t)
+	target := createTarget(t, r)
+
+	handler := http.HandlerFunc(target.targetQueryHandler)
+
+	// malformed odoh query
+	queryBytes := []byte{1, 2, 3}
+	request, err := http.NewRequest(http.MethodPost, queryEndpoint, bytes.NewReader(queryBytes))
+	if err != nil {
+		t.Fatal(err)
+	}
+	request.Header.Add("Content-Type", odohMessageContentType)
+
+	rr := httptest.NewRecorder()
+	handler.ServeHTTP(rr, request)
+
+	if status := rr.Result().StatusCode; status != http.StatusBadRequest {
+		t.Fatal(fmt.Errorf("Result did not yield %d, got %d instead", http.StatusBadRequest, status))
+	}
+}
+
+func TestODoHResolutionWithRealResolver(t *testing.T) {
```

Comment:
This type of dependency shouldn't be in a unit test. Why did we go with 1.1.1.1 here?

Reply:
Likelihood of 1.1.1.1 being unavailable seems extremely low, and I wanted to exercise the `resolve()`: https://github.com/cloudflare/odoh-server-go/blob/master/resolver.go#L47 codepath. I figured that the benefits of having atleast one test for that codepath far outweighed the risk of creating a 1.1.1.1 dependency.

Reply:
I get that, but I think we can exercise that path by mocking things (or wrapping `resolve()` appropriately). Adding a network dependency doesn't seem to be the right answer.

Reply:
re: mocking it, personally I think exercising the networking codepath in `resolve()` is pretty valuable

Reply:
But I can remove it if you think it's fine without.

Reply:
Testing the implementation of `resolve()` isn't something this project should do. We should only be testing that the code that *uses* `resolve` is correct, which is why I suggested mocking it.